suites/apis/googleServiceAccountRemovalTest.js

```diff
@@ -0,0 +1,471 @@
+ * sure we cannot use it after the clean up. Since the validation checks that
+ * there is no external access to the SA, creating a key would prevent us from
+ * testing other removal use cases. This is why the data access is only
+ * checked in the "SA has external access" test.
+ */
+
+
+BeforeSuite(async (google, fence, users) => {
+  await fence.complete.suiteCleanup(google, users);
+});
+
+
+After(async (google, fence, users) => {
+  await fence.complete.suiteCleanup(google, users);
+});
+
+
+AfterSuite(async (google, fence) => {
+  await google.unlockGoogleProject(fence.props.googleProjectDynamic);
+});
+
+
+function checkAndCleanSA() {
+  var fenceCmd = 'fence-create --verbose google-manage-user-registrations';
+  console.log(`Running: ${fenceCmd}`);
+  var jobRes = bash.runCommand(fenceCmd, 'fence');
+  console.log(jobRes);
+  return jobRes;
+}
+
+
+Scenario('SA removal job test: no access removal when SA is valid @reqGoogle', async (fence, users, google, files) => {
+  // test that the clean up job does not remove access to valid SA/projects
+
+  // Setup
+  const googleProject = fence.props.googleProjectDynamic;
+  await fence.complete.forceLinkGoogleAcct(users.user0, googleProject.owner);
+
+  // Register account
+  let registerRes = await fence.do.registerGoogleServiceAccount(
+    users.user0,
+    googleProject,
+    ['QA']
+  );
+  fence.ask.responsesEqual(registerRes, fence.props.resRegisterServiceAccountSuccess);
+
+  // Lock the project
+  lockRes = await google.lockGoogleProject(googleProject);
+  chai.expect(lockRes, 'Could not lock project').to.be.true;
+
+  // Get creds to access data
+  let [pathToKeyFile, keyFullName] = await google.createServiceAccountKeyFile(googleProject);
+
+  // Access data
+  user0AccessQARes = await google.getFileFromBucket(
+    fence.props.googleBucketInfo.QA.googleProjectId,
+    pathToKeyFile,
+    fence.props.googleBucketInfo.QA.bucketId,
+    fence.props.googleBucketInfo.QA.fileName
+  );
+
+  // Delete the creds so that the clean up job will not mark this SA as invalid
+  await google.deleteServiceAccountKey(keyFullName);
+  files.deleteFile(pathToKeyFile);
+
+  // Run clean up job
+  checkAndCleanSA();
+
+  // Get creds to access data
+  [pathToKeyFile, keyFullName] = await google.createServiceAccountKeyFile(googleProject);
+
+  // Access data
+  user0AccessQAResAfter = await google.getFileFromBucket(
+    fence.props.googleBucketInfo.QA.googleProjectId,
+    pathToKeyFile,
+    fence.props.googleBucketInfo.QA.bucketId,
+    fence.props.googleBucketInfo.QA.fileName
+  );
+
+  // Clean up
+  console.log('cleaning up');
+
+  await google.deleteServiceAccountKey(keyFullName);
+  files.deleteFile(pathToKeyFile);
+
+  await fence.do.deleteGoogleServiceAccount(
+    users.user0,
+    googleProject.serviceAccountEmail,
+  );
+
+  await fence.do.unlinkGoogleAcct(
+    users.user0,
+  );
+
+  // Unlock the project
+  let unlockRes = await google.unlockGoogleProject(googleProject);
```

Comment:
what if this fails :grimacing: ? we probably need a big error to the user about how to manually remove the lock